test/test_nn.py

```diff
@@ -188,6 +208,16 @@ def _do_test(self, test_case, module, input):
 class NewCriterionTest(InputVariableMixin, CriterionTest):
     # TODO: check that criterions don't ignore grad_output
 
+    def __init__(self, *args, **kwargs):
+        super(NewCriterionTest, self).__init__(*args, **kwargs)
+        self.check_gradgrad = kwargs.get('check_gradgrad', True)
+
+    def _do_extra_tests(self, test_case, module, input, target):
+        if self.check_gradgrad:
+            params = tuple(x for x in module.parameters())
+            _assertGradAndGradgradChecks(test_case.assertTrue, lambda x, y, *args, **kw: module(x, y),
+                                         (input, target) + params)
```

Comment:
I think you are right that it's necessary; seems better to just keep it unless something changes / we make a mistake somewhere?